servo/connectors/kubernetes.py

```diff
@@ -3009,25 +3010,30 @@ async def apply(self, adjustments: List[servo.Adjustment]) -> None:
         # TODO: Run sanity checks to look for out of band changes
 
     async def raise_for_status(self) -> None:
-        async def _raise_for_task(optimization: BaseOptimization, task: asyncio.Task) -> None:
+        handle_error_tasks = []
+        def _raise_for_task(task: asyncio.Task, optimization: BaseOptimization) -> None:
             if task.done() and not task.cancelled():
                 if exception := task.exception():
-                    await optimization.handle_error(
+                    handle_error_tasks.append(asyncio.create_task(optimization.handle_error(
                         exception, self.config.on_failure
-                    )
+                    )))
 
         tasks = []
         for optimization in self.optimizations:
             task = asyncio.create_task(optimization.raise_for_status())
-            task.add_done_callback(lambda task: _raise_for_task(optimization, task))
+            task.add_done_callback(functools.partial(_raise_for_task, optimization=optimization))
             tasks.append(task)
 
-        for future in asyncio.as_completed(tasks, timeout=self.config.timeout):
+        for future in asyncio.as_completed(tasks, timeout=self.config.timeout.total_seconds()):
             try:
                 await future
             except Exception as error:
                 servo.logger.exception(f"Optimization failed with error: {error}")
 
+        # TODO: first handler to raise will likely interrupt other tasks.
+        #   Gather with return_exceptions=True and aggregate resulting exceptions before raising
+        await asyncio.gather(*handle_error_tasks)
```

Comment:
Why do we want to gather this instead of just letting it run?

Reply:
1. We want to block flow of control so that rollbacks and teardowns are allowed to complete before reporting results to the backend
2. The handle_error task is responsible for communicating the error its handling to the backend. If we allow said tasks to run in the background, the connector would likely [fall through to a more generic error](https://github.com/opsani/servox/blob/faea00e0a6557cd8d7beb0852614d1616e44aef5/servo/connectors/kubernetes.py#L3597) which is undesirable
3. I'm pretty sure exceptions raised by the [handle_error tasks](https://github.com/opsani/servox/blob/faea00e0a6557cd8d7beb0852614d1616e44aef5/servo/connectors/kubernetes.py#L3081) would not be retrieved which would make for some messy logs as well